src/types/workspace.ts

```diff
@@ -29,4 +29,12 @@ export interface IWorkspaceModel extends Model<IWorkspaceSchema> {
     title: string
     workspaceId: IWorkspaceSchema['_id']
   }): Promise<WorkspaceDto | null>
+
+  deleteWorkspace({
+    workspaceId,
+    session,
+  }: {
+    workspaceId: IWorkspaceSchema['_id']
+    session?: ClientSession
```

Comment:
Currently the test environment database doesn't support transactions, so I made the session optional so that we can mock the implementation of this function in the test environment (to run without transaction).

I've added a comment to this to make it clearer!

Can see issue #4503 for details

Reply:
Agreed on this - and we might be able to use our current `mongodb-memory-server-core` [package](https://github.com/nodkz/mongodb-memory-server#replica-set-start) to create a replica set on our DB. Which may help us test for transaction.

However, this branch is too deprecated, will merge it into `form-v2/form-workspaces` first and work on this issue later. 

FRM-976